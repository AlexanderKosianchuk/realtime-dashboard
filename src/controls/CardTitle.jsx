import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  width: 100%;
  padding: 5px 15px 10px;
  margin: 0 0 20px;
  font-size: 1em;
  font-weight: bold;
  border-bottom: 2px solid rgba(30, 30, 30, 0.5);
  text-transform: uppercase;

  @media (min-width: 769px) {
    padding: 10px 25px 15px;
    font-size: 1.6em;
  }
`;

const CardTitle = ({ children }) => <Container>{children}</Container>;

CardTitle.propTypes = {
  children: PropTypes.string.isRequired,
};

export default CardTitle;
