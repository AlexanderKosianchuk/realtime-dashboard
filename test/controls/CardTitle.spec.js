import React from 'react';

import CardTitle from 'controls/CardTitle';

function setup(children = 'title') {
  return mount(<CardTitle>{children}</CardTitle>);
}

describe('Controls Alert', () => {
  let wrapper = setup();

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render text', () => {
    wrapper = setup('Text');

    expect(wrapper).toMatchSnapshot();
  });
});
