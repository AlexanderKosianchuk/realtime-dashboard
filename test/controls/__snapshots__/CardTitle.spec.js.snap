// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Alert should render properly 1`] = `
.c0 {
  width: 100%;
  padding: 5px 15px 10px;
  margin: 0 0 20px;
  font-size: 1em;
  font-weight: bold;
  border-bottom: 2px solid rgba(30,30,30,0.5);
  text-transform: uppercase;
}

@media (min-width:769px) {
  .c0 {
    padding: 10px 25px 15px;
    font-size: 1.6em;
  }
}

<Container>
  <div
    className="c0"
  >
    title
  </div>
</Container>
`;

exports[`Controls Alert should render text 1`] = `
.c0 {
  width: 100%;
  padding: 5px 15px 10px;
  margin: 0 0 20px;
  font-size: 1em;
  font-weight: bold;
  border-bottom: 2px solid rgba(30,30,30,0.5);
  text-transform: uppercase;
}

@media (min-width:769px) {
  .c0 {
    padding: 10px 25px 15px;
    font-size: 1.6em;
  }
}

<Container>
  <div
    className="c0"
  >
    Text
  </div>
</Container>
`;
